Framework/PythonInterface/plugins/algorithms/FitIncidentSpectrum.py

```diff
@@ -0,0 +1,221 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import absolute_import, division, print_function
+import numpy as np
+from scipy import signal, ndimage, interpolate
+from mantid.api import AlgorithmFactory, MatrixWorkspaceProperty, PythonAlgorithm
+from mantid.kernel import Direction, StringListValidator
+from mantid.simpleapi import CreateWorkspace, Rebin, SplineSmoothing, AnalysisDataService
+
+
+class FitIncidentSpectrum(PythonAlgorithm):
+    _input_ws = None
+    _output_ws = None
+    _scipy_not_old = hasattr(interpolate.UnivariateSpline, "derivative")
+    # check if scipy version is greater than 0.12.1 i.e it has the derivative function
+
+    def category(self):
+        return 'Diffraction\\Fitting'
+
+    def name(self):
+        return 'FitIncidentSpectrum'
+
+    def summary(self):
+        return 'Calculate a fit for an incident spectrum using different methods. ' \
+               'Outputs a workspace containing the functionalized fit and its first ' \
+               'derivative.'
+
+    def seeAlso(self):
+        return [""]
+
+    def version(self):
+        return 1
+
+    def PyInit(self):
+        self.declareProperty(
+            MatrixWorkspaceProperty('InputWorkspace', '',
+                                    direction=Direction.Input,
+                                    ),
+            doc='Incident spectrum to be fit.')
+
+        self.declareProperty(
+            MatrixWorkspaceProperty('OutputWorkspace', '',
+                                    direction=Direction.Output),
+            doc='Output workspace containing the fit and it\'s first derivative.')
+
+        self.declareProperty(
+            name='BinningForCalc',
+            defaultValue='',
```

Comment:
Can I suggest we don't use a string here and use an [`FloatArrayProperty`](https://www.mantidproject.org/Declaring_Array_Properties)?. It will then match the same input type as `Rebin`
We also have a class called `RebinParamsValidator` that we could then use here to verify that the binning parameters are correct. You can use it by adding a `validator = RebinParamsValidator()` argument to the `declareProperty` call. I think this would remove the need to do the manual check in `validateInputs` for both binning inputs.